test/e2e/tests/test_notebook_instance.py

```diff
@@ -0,0 +1,169 @@
+        logging.error(
+            f"SageMaker could not find a Notebook Instance with the name {notebook_instance_name}. Error {error}"
+        )
+        return None
+
+
+def get_notebook_instance_sagemaker_status(notebook_instance_name: str):
+    notebook_instance = get_notebook_instance(notebook_instance_name)
+    assert notebook_instance is not None
+    return notebook_instance["NotebookInstanceStatus"]
+
+
+def get_notebook_instance_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert resource is not None
+    assert "notebookInstanceStatus" in resource["status"]
+    return resource["status"]["notebookInstanceStatus"]
+
+
+@pytest.mark.canary
+@service_marker
+class TestNotebookInstance:
+    def _wait_resource_notebook_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_notebook_instance_resource_status,
+            reference,
+        )
+
+    def _wait_sagemaker_notebook_status(
+        self,
+        notebook_instance_name: str,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_notebook_instance_sagemaker_status,
+            notebook_instance_name,
+        )
+
+    def _assert_notebook_status_in_sync(
+        self, notebook_instance_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_notebook_status(
+                notebook_instance_name, expected_status
+            )
+            == self._wait_resource_notebook_status(reference, expected_status)
+            == expected_status
+        )
+    def testCreateUpdateDelete(self,notebook_instance):
+        (reference, resource, spec) = notebook_instance
+        assert k8s.get_resource_exists(reference)
+        
+        #Create the resource and verify that its Pending
+        notebook_instance_name = resource["spec"].get("notebookInstanceName",None)
+        assert notebook_instance_name is not None
+
+        notebook_description = get_notebook_instance(notebook_instance_name)
+        assert notebook_description["NotebookInstanceStatus"] == "Pending"
+
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+        self._assert_notebook_status_in_sync(notebook_instance_name,reference,"Pending")
+
+        #wait for the resource to go to the InService state and make sure the operator is synced with sagemaker.
+        self._assert_notebook_status_in_sync(notebook_instance_name,reference,"InService")
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        #Update test
```

Comment:
Could you test update on a field that is not a part of the create request such as - DisassociateLifecycleConfig

Reply:
Also, could you move this to a separate test such as here - https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/test/e2e/tests/test_endpoint.py#L304

Reply:
Handled in #82